lib/internal/perf/observe.js

```diff
@@ -0,0 +1,352 @@
+'use strict';
+
+const {
+  ArrayFrom,
+  ArrayIsArray,
+  ArrayPrototypeFilter,
+  ArrayPrototypeIncludes,
+  ArrayPrototypePush,
+  ArrayPrototypeSlice,
+  ArrayPrototypeSort,
+  ObjectDefineProperties,
+  ObjectFreeze,
+  ObjectKeys,
+  SafeMap,
+  SafeSet,
+  Symbol,
+} = primordials;
+
+const {
+  observerCounts,
+  installGarbageCollectionTracking,
+  removeGarbageCollectionTracking,
+  setupObservers,
+  constants: {
+    NODE_PERFORMANCE_ENTRY_TYPE_GC,
+    NODE_PERFORMANCE_ENTRY_TYPE_FUNCTION,
+    NODE_PERFORMANCE_ENTRY_TYPE_HTTP2,
+    NODE_PERFORMANCE_ENTRY_TYPE_HTTP,
+  }
```

Comment:
nit: ASCII order and trailing comma
```suggestion
  constants: {
    NODE_PERFORMANCE_ENTRY_TYPE_GC,
    NODE_PERFORMANCE_ENTRY_TYPE_FUNCTION,
    NODE_PERFORMANCE_ENTRY_TYPE_HTTP2,
    NODE_PERFORMANCE_ENTRY_TYPE_HTTP,
  },
  installGarbageCollectionTracking,
  observerCounts,
  removeGarbageCollectionTracking,
  setupObservers,
```